### DISCUSSION TITLE
Consulta consigna ejercicio 11

### DISCUSSION THREAD

**Discussion by vlzJuan** [INGCOM-UNRN-P1/cursada-2023#51] [Preguntas y respuestas] [answered]
Buenas noches. Tengo una duda respecto a lo solicitado en el ejercicio 11.

La consigna nos pide que la función 'retorne' el número incrementado en 1 hasta el límite. Y luego que "Cuando `i == limite`, la función retornará `0`."

Aca, el primer 'retornar' tiene que ver con imprimir en pantalla cada 'n', entre 'i' y 'limite', en incrementos de a 1, sin incluir el limite. El segundo 'retornar' se refiere al valor devuelto por la función "int contador_circular(int i, int limite)". ¿Es mi interpretación correcta?

Adjunto mi versión actual del código, que debería corresponderse al caso interpretado como expliqué:

<details>
  <summary>Spoiler</summary>

```C
/*
Ejercicio 11
Juan Manuel Valenzuela
Comision 1 - 2023
Programacion 1 - Ingenieria en computacion
Usuario GitHub: vlzJuan
*/

#include <stdio.h>

int contador_circular(int i, int limite){
    int step=1; // 'Paso' del incremento para el for. Con esto me aseguro que funcione para cualquier limite.
    if (i>limite){
        step=-1;
    }
    for (int n=i; n!=limite; n+=step){
        printf("%d\n", n); //Imprimo cada n hasta llegar a limite
    }

    return 0;
}

int main()
{
    int i, limite;
    printf("Ingrese dos numeros enteros, separados por un espacio: ");
    scanf("%d %d", &i, &limite);
    contador_circular(i, limite);
}
// gcc -o ejercicio11 ejercicio11.c
// ./ejercicio11.exe
```

</details>

Saludos

**Comment by martinvilu** [comment, marked as answer]
La función no necesita de un lazo, el lazo va en el main, ya que el proposito de la misma no es mostrar el número sino mantenerlo entre `0` y `limite`; nada mas

**Reply by vlzJuan** [reply]
Buenas! Con la consigna reinterpretada, mi código queda así:
<details>
  <summary>Spoiler</summary>

```C
/*
Ejercicio 11
Juan Manuel Valenzuela
Comision 1 - 2023
Programacion 1 - Ingenieria en computacion
Usuario GitHub: vlzJuan
*/

#include <stdio.h>

int contador_circular(int i, int limite){

    if (i==limite){
        return 0;
    }
    else{
        return i+1;
    }    
}

int main()
{
    int i, limite;
    printf("Ingrese dos numeros enteros, separados por un espacio: ");
    scanf("%d %d", &i, &limite);
    while(i!=limite){
        i=contador_circular(i,limite);
        printf("%d\n", i);
    }

}
// gcc -o ejercicio11 ejercicio11.c
// ./ejercicio11.exe

```

</details>

¿A esto se refiere con 'incluir el loop en el main?
Saludos
